huaweicloud/resource_huaweicloud_dms_rabbitmq_instance.go

```diff
@@ -0,0 +1,408 @@
+	d.Set("used_storage_space", v.UsedStorageSpace)
+	d.Set("connect_address", v.ConnectAddress)
+	d.Set("manegement_connect_address", v.ManagementConnectAddress)
+	d.Set("port", v.Port)
+	d.Set("status", v.Status)
+	d.Set("resource_spec_code", v.ResourceSpecCode)
+	d.Set("user_id", v.UserID)
+	d.Set("user_name", v.UserName)
+	d.Set("type", v.Type)
+	d.Set("access_user", v.AccessUser)
+
+	// set tags
+	engine := "rabbitmq"
+	if resourceTags, err := tags.Get(dmsV2Client, engine, d.Id()).Extract(); err == nil {
+		tagmap := utils.TagsToMap(resourceTags.Tags)
+		if err := d.Set("tags", tagmap); err != nil {
+			return fmt.Errorf("error saving tags to state for dms rabbitmq instance (%s): %s", d.Id(), err)
+		}
+	} else {
+		log.Printf("[WARN] error fetching tags of dms rabbitmq instance (%s): %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func resourceDmsRabbitmqInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	dmsV2Client, err := config.DmsV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating HuaweiCloud dms instance client: %s", err)
+	}
+
+	//lintignore:R019
+	if d.HasChanges("name", "description", "maintain_begin", "maintain_end",
+		"security_group_id", "public_ip_id", "enterprise_project_id") {
+		description := d.Get("description").(string)
+		updateOpts := instances.UpdateOpts{
+			Description:         &description,
+			MaintainBegin:       d.Get("maintain_begin").(string),
+			MaintainEnd:         d.Get("maintain_end").(string),
+			SecurityGroupID:     d.Get("security_group_id").(string),
+			EnterpriseProjectID: d.Get("enterprise_project_id").(string),
+		}
+
+		if d.HasChange("name") {
+			updateOpts.Name = d.Get("name").(string)
+		}
+
+		if d.HasChange("public_ip_id") {
+			if v, ok := d.GetOk("public_ip_id"); ok {
+				enablePublicIP := true
+				updateOpts.EnablePublicIP = &enablePublicIP
+				updateOpts.PublicIpID = v.(string)
+			} else {
+				enablePublicIP := false
+				updateOpts.EnablePublicIP = &enablePublicIP
+			}
+		}
+
+		err = instances.Update(dmsV2Client, d.Id(), updateOpts).Err
+		if err != nil {
+			return fmt.Errorf("error updating HuaweiCloud Dms rabbitmq Instance: %s", err)
+		}
+	}
+
+	if d.HasChange("tags") {
+		// update tags
+		engine := "rabbitmq"
+		tagErr := utils.UpdateResourceTags(dmsV2Client, d, engine, d.Id())
+		if tagErr != nil {
+			return fmt.Errorf("error updating tags of dms rabbitmq instance:%s, err:%s", d.Id(), tagErr)
+		}
+	}
+
+	return resourceDmsRabbitmqInstanceRead(d, meta)
+}
+
+func resourceDmsRabbitmqInstanceDelete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	dmsV2Client, err := config.DmsV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating HuaweiCloud dms instance client: %s", err)
+	}
+
+	_, err = instances.Get(dmsV2Client, d.Id()).Extract()
+	if err != nil {
+		return CheckDeleted(d, err, "instance")
+	}
```

Comment:
it's unnecessary to Get the instance before deleting it.

Reply:
fixed